src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -247,11 +254,24 @@ protected void AddSecrets(TransactionBuildContext context)
         }
 
         /// <summary>
-        /// Find the next available change address.
+        /// If a ChangeAddress was set in the options send change there, otherwise find the next available change address.
         /// </summary>
         /// <param name="context">The context associated with the current transaction being built.</param>
         protected void FindChangeAddress(TransactionBuildContext context)
         {
+            if (context.BuildOptions.ChangeAddress != null)
+            {
+                // Ensure the given ChangeAddress is from the same wallet. 
+                Wallet wallet = this.walletManager.GetWalletByName(context.AccountReference.WalletName);
+                if (!wallet.ContainsAddress(context.BuildOptions.ChangeAddress))
+                {
+                    throw new WalletException("Change address given does not belong to this wallet");
+                }
+
+                context.TransactionBuilder.SetChange(context.BuildOptions.ChangeAddress.ScriptPubKey);
+                return;
+            }
+
             // Get an address to send the change to.
```

Comment:
comment would be redundant here if you methodise the two things within FindChangeAddress